web/components/templates/requests/components/chatComponent/ChatMessageTopBar.tsx

```diff
@@ -32,8 +32,53 @@ interface ChatMessageTopBarProps {
   deleteMessage: (_index: number) => void;
   popoverOpen: boolean;
   setPopoverOpen: (_open: boolean) => void;
+  onAddText?: () => void;
+  onAddImage?: () => void;
 }
 
+const ROLE_OPTIONS = [
+  { value: "user", label: "User" },
+  { value: "assistant", label: "Assistant" },
+  { value: "system", label: "System" },
+  { value: "tool", label: "Tool" },
+  { value: "image", label: "Image" },
+] as const;
+
+const getDropdownItems = (
+  messageRole: string,
+  handlers: {
+    addToolCall: () => void;
+    onAddText?: () => void;
+    onAddImage?: () => void;
+  }
+) => {
+  const items = [];
+
+  if (messageRole === "assistant") {
+    items.push({
+      label: "Add Tool Call",
+      onClick: handlers.addToolCall,
+    });
+  }
+
+  if (messageRole === "user") {
+    if (handlers.onAddText) {
+      items.push({
+        label: "Add Text",
+        onClick: handlers.onAddText,
+      });
+    }
+    if (handlers.onAddImage) {
+      items.push({
+        label: "Add Image",
+        onClick: handlers.onAddImage,
+      });
+    }
+  }
```

Comment:
this is just me being nitpicky but not a huge fan of `push`ing stuff into an array and mutating it - can we just do it while instantiating something like this
```suggestion
const items = [
  ...(messageRole === "assistant"
    ? [{ label: "Add Tool Call", onClick: handlers.addToolCall }]
    : []),

  ...(messageRole === "user" && handlers.onAddText
    ? [{ label: "Add Text", onClick: handlers.onAddText }]
    : []),

  ...(messageRole === "user" && handlers.onAddImage
    ? [{ label: "Add Image", onClick: handlers.onAddImage }]
    : []),
];
```